src/TextField.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module TextField
+ */
+class TextField extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value,
+			error: props.error
+		};
```

Comment:
ok. I think @adamd wanted to try to have a charcounter sketched out in this branch. so I'll fix this up depending on that.